rocketpy/simulation/flight.py

```diff
@@ -607,6 +606,20 @@ def __init__(
         self.name = name
         self.equations_of_motion = equations_of_motion
 
+        # Controller initialization
+        self._controllers = self.rocket._controllers[:]
+        if self._controllers:
+            if self.time_overshoot == True:
+                warnings.warn(
+                    "time_overshoot is set to True, but controllers are present. "
+                    "Controllers will not work properly. "
+                    "Consider setting time_overshoot=False in the Flight object "
+                    "to use controllers."
+                )
+            # reset controllable object to initial state (only airbrakes for now)
+            for air_brakes in self.rocket.air_brakes:
+                air_brakes._reset()
+
```

Comment:
This code block (from "# Controller initialization" to "air_brakes._reset") is too long. To simplify the `__init__` method, I would create a new `__init_controllers` method and invoke it right here.

Also, regarding your message "Controllers will not work properly", I understand that the controllers will not work at all if you set time_overshoot to true. Is that correct?

What is better:
1. Throwing a warning that the simulation will not work as expected, but run the simulation anyways. 
2. Throwing a warning that the time_overshoot is being changed to `False` and then the simulation?

Reply:
I prefer the second option. However, is forcing a parameter to be something a good practice?
In this case it seems to make sense, but maybe it is worth the discussion.

Reply:
It is not a good practice if you don't let the user know it. But since we are throwing an warning, I believe it is more than OK.